### PR TITLE
EOS-18920: Patch-2: Config validations added

### PR DESCRIPTION
**Pull request by divya-vijay**
Changes updated over the same branch for PR #1060 , will rebase once that is merged.

### CODE REVIEW

File: api/python/provisioner/commands/validator/node.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        missing_params = []
+        for param, value in params.items():
+            if value == UNCHANGED and param not in self._optional_param:
+                missing_params.append(param)
+        if len(missing_params) > 0:
+            logger.error(
+                f"{missing_params} cannot be left empty or blank.. "
+                "These are mandatory to configure the setup."
+            )
+            raise ValueError(f"Mandatory param missing {missing_params}")
```
Comment (dmitrydb) [inline api/python/provisioner/commands/validator/node.py:116]: There are no new line here after the last code line.

File: api/python/provisioner/commands/validator/release.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        missing_params = []
+        for param, value in params.items():
+            if value == UNCHANGED and param not in self._optional_param:
+                missing_params.append(param)
+        if len(missing_params) > 0:
+            logger.error(
+                f"{missing_params} cannot be left empty or blank.. "
+                "These are mandatory to configure the setup."
+            )
+            raise ValueError(f"Mandatory param missing {missing_params}")
```
Comment (dmitrydb) [inline api/python/provisioner/commands/validator/release.py:47]: There are no new line here after the last code line.

File: api/python/provisioner/commands/validator/release.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    ReleaseParams
+)
+from provisioner.values import UNCHANGED
+from provisioner.vendor import attr
+
+logger = logging.getLogger(__name__)
+
+
+@attr.s(auto_attribs=True)
+class ReleaseParamsValidation:
```
Comment (dmitrydb) [inline api/python/provisioner/commands/validator/release.py:32]: I think it is better to align class name with other validation classes: `ReleaseParamsValidator`

File: api/python/provisioner/commands/validator/network.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        missing_params = []
+        for param, value in params.items():
+            if value == UNCHANGED and param not in self._optional_param:
+                missing_params.append(param)
+        if len(missing_params) > 0:
+            logger.error(
+                f"{missing_params} cannot be left empty or blank.. "
+                "These are mandatory to configure the setup."
+            )
+            raise ValueError(f"Mandatory param missing {missing_params}")
```
Comment (dmitrydb) [inline api/python/provisioner/commands/validator/network.py:51]: There are no new line here after the last code line.

File: api/python/provisioner/commands/validator/enclosure.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+                params['type'] == 'RBOD'
+            ):
+                missing_params.append(param)
+
+        if len(missing_params) > 0:
+            logger.error(
+                f"{missing_params} cannot be left empty or blank.. "
+                "These are mandatory to configure the setup."
+            )
+            raise ValueError(f"Mandatory param missing {missing_params}")
```
Comment (dmitrydb) [inline api/python/provisioner/commands/validator/enclosure.py:63]: There are no new line here after the last code line.

File: api/python/provisioner/commands/validator/config_sections.py [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+                    )
+
+                else:
+                    logger.info(
+                        "Success: Node Validation done. "
+                        f"Applying config for {node_count}-node setup"
+                    )
+
+        except Exception as exc:
+            raise ValueError(f"Config Failed to apply: {str(exc)}")
```
Comment (dmitrydb) [inline api/python/provisioner/commands/validator/config_sections.py:125]: There are no new line here after the last code line.

**Review by dmitrydb — CHANGES_REQUESTED** [review]
I added minor comments

File: api/python/provisioner/commands/validator/node.py (hunk 1 above)
Comment (divya-vijay) [reply api/python/provisioner/commands/validator/node.py:116]: @dmitrydb added !

File: api/python/provisioner/commands/validator/release.py (hunk 2 above)
Comment (divya-vijay) [reply api/python/provisioner/commands/validator/release.py:47]: Done.

File: api/python/provisioner/commands/validator/release.py (hunk 3 above)
Comment (divya-vijay) [reply api/python/provisioner/commands/validator/release.py]: yes, edited. Thanks

File: api/python/provisioner/commands/validator/enclosure.py (hunk 5 above)
Comment (divya-vijay) [reply api/python/provisioner/commands/validator/enclosure.py:63]: Done!

File: api/python/provisioner/commands/validator/config_sections.py (hunk 6 above)
Comment (divya-vijay) [reply api/python/provisioner/commands/validator/config_sections.py:125]: Done

**Review by dmitrydb — APPROVED** [review]
Looks good for me. I approved

**Comment by dmitrydb** [comment]
@divya-vijay Github shows conflicts with `pre-cortx-1.0` branch. Please, merged `pre-cortx-1.0` branch to this PR and resolve conflicts

**Comment by divya-vijay** [comment]
> @divya-vijay Github shows conflicts with `pre-cortx-1.0` branch. Please, merged `pre-cortx-1.0` branch to this PR and resolve conflicts

@dmitrydb I think it is because, this PR was raised on top of another branch, which is yet to be merged with pre-cortx. Once that is done, I can rebase origin and update this PR.

**Review by ypise — APPROVED** [review]

**Review by ypise — APPROVED** [review]

**Review by mazinamdar — APPROVED** [review]
looks good

**Pull request merged**